src/task7/task7

```diff
@@ -0,0 +1,34 @@
+function arrayToList(arr) {
+    let list = {};
+    for (let i = arr.length - 1; i >= 0; i--) {
+        list.value = arr[i];
+        list.rest = arr.length > 1 ? arrayToList(arr.slice(1)) : null;
+    }
+    return list;
+}
+
+function listToArray(list) {
+    let array = [];
+    for (; list; list = list.rest) {
+        array.push(list.value);
+    }
+    return array;
+}
+
+function prepend(list, value) {
+    return list = { value: value, rest: list };
+}
+
+function nth(list, n) {
+    let array = [];
+    for (; array.length <= n; list = list.rest) {
```

Comment:
Don't change the initial value

Reply:
Hold on.
You don't need this function to complete nth2... 
**nth** should be enough.
Recursion and ternary is the key to success.

Reply:
yeah, I meant nth2 (recursive), but inattentive is my middle name
I don't still know how to get rid of that i=0, doesn't seem elegant